src/ifp/src/InitFloorplan.cc

```diff
@@ -160,32 +160,22 @@ static int divCeil(int dividend, int divisor)
 
 void InitFloorplan::initFloorplan(const odb::Rect& die,
                                   const odb::Rect& core,
-                                  const std::string& site_name)
+                                  const std::vector<odb::dbSite*>& user_sites)
 {
   Rect die_area(snapToMfgGrid(die.xMin()),
                 snapToMfgGrid(die.yMin()),
                 snapToMfgGrid(die.xMax()),
                 snapToMfgGrid(die.yMax()));
   block_->setDieArea(die_area);
-  auto sites = getSites();
-  bool site_found_in_instances
-      = std::any_of(sites.begin(), sites.end(), [&](const auto& site) {
-          return site_name == site->getName();
-        });
-
-  if (!site_found_in_instances && !site_name.empty()) {
-    dbSite* site_found = findSite(site_name.c_str());
-    if (site_found != nullptr) {
-      // this indeed happens for unithd and unit only in PAD.
-      // logger_->warn(IFP,
-      //               434,
-      //               "Number of instances found {}, but Site {} you passed was
-      //               " "not in them.", sites.size(), site_name);
-      sites.insert(site_found);
-    } else {
-      logger_->warn(IFP, 41, "Site {} not found in Database.", site_name);
-    }
+  std::set<odb::dbSite*> sites = getSites();
+  sites.insert(user_sites.begin(), user_sites.end());
+
+  // Handle duplicated sites
+  std::map<std::string, odb::dbSite*> sites_by_name;
+  for (auto* site : sites) {
+    sites_by_name[site->getName()] = site;
   }
```

Comment:
You can't have two different sites with the same name - odb would error.  Is this necessary?

Reply:
Which testcase?  It looks like each LEF (dbLib) is allowed to define its own SITEs.

Reply:
init_floorplan9.tcl:

If you modify the script:
```
# init_floorplan called after a voltage domain is specified
source "helpers.tcl"
read_lef sky130hd/sky130hd.tlef
read_lef sky130hd/sky130_fd_sc_hd_merged.lef
read_liberty sky130hd/sky130_fd_sc_hd__tt_025C_1v80.lib
read_verilog reg2.v
link_design top

#create_voltage_domain TEMP_ANALOG -area {27 27 60 60}
#initialize_floorplan -die_area "0 0 150 150" \
#  -core_area "20 20 130 130" \
#  -site unithd
create_voltage_domain TEMP_ANALOG -area {33.58 32.64 64.86 62.56}
initialize_floorplan -die_area "0 0 155.48 146.88" \
  -core_area "18.4 16.32 137.08 130.56" 
  
set def_file [make_result_file init_floorplan9.def]
write_def $def_file
diff_files init_floorplan9.defok $def_file


foreach lib [[ord::get_db] getLibs] {
  foreach site [$lib getSites] {
    puts "[$lib getName] [$site getName] $site"
  }
}
```

It will output 2 unithd sites with a unique pointer to each (even if they are the same).
```
sky130hd unithd _d00fcd209d550000_p_odb__dbSite
sky130hd unithddbl _f00fcd209d550000_p_odb__dbSite
sky130_fd_sc_hd_merged unithd _c0b3c2209d550000_p_odb__dbSite
```

Reply:
In lefin::macro it copies the site, see https://github.com/The-OpenROAD-Project/OpenROAD/blob/97bb79043c39cac424604e50450f2df9484e4ded/src/odb/src/lefin/lefin.cpp#L1222

@osamahammad21 implemented [this code](https://github.com/The-OpenROAD-Project/OpenROAD/commit/d794f6dd75d39a2c75de2dd762fea927c058dfa8) so I'm not quite sure what the motivation was.  He is off for Eid so I'll approve and we can revisit when he's back.